src/calculator.erl

```diff
@@ -0,0 +1,36 @@
+-module(calculator).
+-export([add/2, subtract/2, multiply/2, divide/2, unused_function/0, main/0]).
+
+% Function to add two integers
+-spec add(integer(), integer()) -> integer().
+add(X, Y) ->
+  X + Y + 0.5. % Returns a float instead of an integer
```

Comment:
calculator.erl:5: Invalid type specification for function calculator:add/2. The success typing is 
          (number(), number()) -> float()